### DISCUSSION TITLE
Field Resolver returns undefined

### DISCUSSION THREAD

**Discussion by didzis1** [MichalLytek/type-graphql#1115] [Q&A] [answered]
I have an ObjectType that I have created with Typegoose + TypeGraphQL:
![image](https://user-images.githubusercontent.com/53485411/141934271-f2f781c3-5138-44c4-b1e4-e05c57fea534.png)

workoutAmount should return the length of workouts array, but `this.workouts` returns undefined. If I `console.log(this)`, then I see this:
![image](https://user-images.githubusercontent.com/53485411/141934235-40ecc842-f2de-4db7-af93-00ff5550ae5e.png)

Why is this happening?

**Comment by MichalLytek** [comment, marked as answer]
Search for mongoose/typegoose issues and checkout examples.
You need to convert mongoose document into a plain JS object while returning from resolvers.

**Reply by didzis1** [reply]
Thanks a lot for the hint! I managed to get it to work by implementing a middleware, just like in the TypeGraphQL Typegoose examples -> https://github.com/MichalLytek/type-graphql/blob/master/examples/typegoose/typegoose-middleware.ts
